mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -210,6 +214,25 @@ class DecisionTreeClassificationModel private[ml] (
     rootNode.predictImpl(features).prediction
   }
 
+  /** @group setParam */
+  @Since("3.0.0")
+  def setLeafCol(value: String): this.type = set(leafCol, value)
+
+  @Since("3.0.0")
+  def predictLeaf(features: Vector): Double = predictLeafImpl(features)
```

Comment:
Can this go in a superclass? Is there any need to separate predictLeaf from predictLeafImpl?
Likewise can setLeafCol go in a superclass to avoid repeating it?

Reply:
@srowen I prefer not to move `setLeafCol` into the superclass, since it looks like the mllib's convention, such as `setVarianceCol` in `DecisionTreeRegressionModel`.